mwedittypes/tokenizer.py

```diff
@@ -35,7 +38,17 @@ def get_whitespace(self, text):
 
     def get_words(self, text):
         #This extracts words inclusive of those with hyphens and apostrophes
-        word_list = re.findall(r"[\w'-]+",text)
+        if self.lang in NON_WHITESPACE_LANGUAGES:
+            word_list = re.findall(r"[\w]", text)
+
+        else:
+            #Get words with hyphens and apostrophe
+            #\b at the start and end of the word ensures there is a word boundary before a character followed by a hyphen or apostrophe
+            hyphenated_apos_words = re.findall(r"\b\w*[-']\w*\b", text)
+
+            #Remove hyphenated words
+            text = re.sub(r"\b\w*[-']\w*\b",'',text)
+            word_list = re.findall(r"[\w]+",text) + hyphenated_apos_words
```

Comment:
Alright. I’d include it @geohci

Reply:
done - thanks! just as a note to myself, when the apostrophe or hyphen is at the end of the word, it's not captured. e.g., "word-" and "word" both end up as "word" in the words dictionary. I hadn't considered this but I think that's actually okay because punctuation still catches the hyphen/apostrophe and we don't e.g., include trailing commas or other punctuation at the start/end of words.